python_modules/dagster/dagster/_core/event_api.py

```diff
@@ -146,3 +146,155 @@ def __new__(
+            events cursor will result in a significant performance gain when run against
+            a SqliteEventLogStorage implementation (which is run-sharded)
+        after_timestamp (Optional[float]): Filter parameter such that only event records for
+            events with timestamp greater than the provided value are returned.
+        before_timestamp (Optional[float]): Filter parameter such that only event records for
+            events with timestamp less than the provided value are returned.
+        storage_ids (Optional[Sequence[int]]): Filter parameter such that only event records for
+            the given storage ids are returned.
+        tags (Optional[Mapping[str, Union[str, Sequence[str]]]]): Filter parameter such that only
+            events with the given event tags are returned
+    """
+
+    def __new__(
+        cls,
+        asset_key: Optional[AssetKey] = None,
+        asset_partitions: Optional[Sequence[str]] = None,
+        after_cursor: Optional[Union[int, RunShardedEventsCursor]] = None,
+        before_cursor: Optional[Union[int, RunShardedEventsCursor]] = None,
+        after_timestamp: Optional[float] = None,
+        before_timestamp: Optional[float] = None,
+        storage_ids: Optional[Sequence[int]] = None,
+        tags: Optional[Mapping[str, Union[str, Sequence[str]]]] = None,
+    ):
+        return super(AssetRecordsFilter, cls).__new__(
+            cls,
+            asset_key=check.opt_inst_param(asset_key, "asset_key", AssetKey),
+            asset_partitions=check.opt_nullable_sequence_param(
+                asset_partitions, "asset_partitions", of_type=str
+            ),
+            after_cursor=check.opt_inst_param(
+                after_cursor, "after_cursor", (int, RunShardedEventsCursor)
+            ),
+            before_cursor=check.opt_inst_param(
+                before_cursor, "before_cursor", (int, RunShardedEventsCursor)
+            ),
+            after_timestamp=check.opt_float_param(after_timestamp, "after_timestamp"),
+            before_timestamp=check.opt_float_param(before_timestamp, "before_timestamp"),
+            storage_ids=check.opt_nullable_sequence_param(storage_ids, "storage_ids", of_type=int),
+            tags=check.opt_nullable_mapping_param(tags, "tags", key_type=str),
+        )
+
+    def to_event_records_filter(self, event_type: DagsterEventType) -> EventRecordsFilter:
+        return EventRecordsFilter(
+            event_type=event_type,
+            asset_key=self.asset_key,
+            asset_partitions=self.asset_partitions,
+            after_cursor=self.after_cursor,
+            before_cursor=self.before_cursor,
+            after_timestamp=self.after_timestamp,
+            before_timestamp=self.before_timestamp,
+            storage_ids=self.storage_ids,
+            tags=self.tags,
+        )
+
+
+@whitelist_for_serdes
+class RunStatusEventRecordsFilter(
```

Comment:
maybe `RunStatusChangeEvent`? theres some subtlety around these events representing the change of run status